pdfminer/layout.py

```diff
@@ -723,8 +742,8 @@ def analyze(self, laparams):
         for obj in empties:
             obj.analyze(laparams)
         textboxes = list(self.group_textlines(laparams, textlines))
-        if -1 <= laparams.boxes_flow and laparams.boxes_flow <= +1 \
-                and textboxes:
+        if laparams.boxes_flow is not None and \
+                -1 <= laparams.boxes_flow <= +1 and textboxes:
```

Comment:
This can be more simple:

```python
if laparams.boxes_flow is None:
    ....
elif -1 <- laparams.boxes_flow <= 1 and textboxes:
    ...
else:
    # should never happend due to _validate()
```

You can even do:

```python
if laparams.boxes_flow is None:
    ...
else:
    ...
```

if both sorting methods also work on an empty list.